Add documentation about different installations options

ajfriesen opened the pull request:
**Description:**

A lot of people do not know what the real difference between all this installations methods are.
Then there are Hass.io Add-Ons which get people confused as well.
I was also in the place and it did take long to understand that. (Maybe I am still not there ;-))

This could also go to the FAQ as well, just started a draft.

The graphics are really not fancy but will hopefully make something more clear.
Made with yed. Added the hassio_on_hassos.graphml file for later editing.

What do you think?

**Pull request in [home-assistant](https://github.com/home-assistant/home-assistant) (if applicable):** home-assistant/home-assistant#<home-assistant PR number goes here>

## Checklist:

- [ ] Branch: `next` is for changes and new documentation that will go public with the next [home-assistant](https://github.com/home-assistant/home-assistant) release. Fixes, changes and adjustments for the current release should be created against `current`.
- [ ] The documentation follows the [standards][standards].

[standards]: https://developers.home-assistant.io/docs/documentation_standards.html

ajfriesen committed:
Add documentation about different installations options

klaasnicolaas commented on source/_docs/installation.markdown:
```diff
@@ -131,3 +131,124 @@ For more details about `hass`, please refer to the [tools section](/docs/tools/h
+It is almost the same like Hass.io on HassOS but you do run Hass.io on your own operating system of choice instead of HassOS.
+
+You will get:
+- full Home Assistant
```
What do you mean with this?

ajfriesen replied:
Each installation method will give you home assistant with all features.

Often people do not understand that the hassio add-ons are not home assistant functionality itself, it is just more or less preconfigured software to work with home assistant.

frenck commented:
I'm going to assign myself as reviewer, quick glance, there are quite a bit of incorrect statements in here and spotted some spelling/grammar errors.

DubhAd commented:
IMO, this kind of guide would be better done on its own page.

frenck commented on source/_docs/installation.markdown:
```diff
@@ -131,3 +131,124 @@ For more details about `hass`, please refer to the [tools section](/docs/tools/h
+
+<p class='note'>
+Add-Ons provide an easy way to add more functionality to your setup through the gui.
+Add-Ons in Hass.io are just docker containers containing some software.
```
Basically, correct, nevertheless, they do have more integrations possibilities in comparison to plain Docker.

```suggestion
Add-ons in Hass.io are essentially just docker containers containing some software.
```

frenck commented on source/_docs/installation.markdown:
```diff
@@ -131,3 +131,124 @@ For more details about `hass`, please refer to the [tools section](/docs/tools/h
+
+You will get:
+- full Home Assistant
+- Hass.io Add-Ons (some may not work properly)
```
`(some may not work properly)` this is actually incorrect. Add-ons may have an architecture limitation, so do docker containers or any other piece of software you'd like to install on a certain computer.

ajfriesen replied:
I just added this comment as I saw this: 

```
When you use this installation method, the core SSH add-on may not function correctly. If that happens, use the community SSH add-on. Some of the documentation might not work for your installation either.
```
[Source](https://www.home-assistant.io/hassio/installation/#alternative-install-on-generic-linux-server)

I do not know how deep the add-ons are depending on hassio and therefore on hassOS.

If there is no other dependency than the architecture (for example there is only an arm6 container available but not for arm7) then we should not mention that.

frenck commented on source/_docs/installation.markdown:
```diff
@@ -131,3 +131,124 @@ For more details about `hass`, please refer to the [tools section](/docs/tools/h
+
+## {% linkable_title Characteristics of different install methods for Home Assistant %}
+
+You always get the full Home Assistant expirience despite the installation method you choose to use.
```
```suggestion
You always get the full Home Assistant experience despite the installation method you choose to use.
```

frenck commented on source/_docs/installation.markdown:
```diff
@@ -131,3 +131,124 @@ For more details about `hass`, please refer to the [tools section](/docs/tools/h
+## {% linkable_title Characteristics of different install methods for Home Assistant %}
+
+You always get the full Home Assistant expirience despite the installation method you choose to use.
+How you would like to run Home Assistant is a choice of taste, the setup you already have, knowlede and what you want to accomplish with home assistant.
```
```suggestion
How you would like to run Home Assistant is a choice of taste, the setup you already have, knowledge and what you want to accomplish with Home Assistant.
```

frenck commented on source/_docs/installation.markdown:
```diff
@@ -131,3 +131,124 @@ For more details about `hass`, please refer to the [tools section](/docs/tools/h
+How you would like to run Home Assistant is a choice of taste, the setup you already have, knowlede and what you want to accomplish with home assistant.
+
+<p class='note'>
+Add-Ons provide an easy way to add more functionality to your setup through the gui.
```
```suggestion
Add-ons provide an easy way to add more functionality to your setup through the GUI.
```

frenck commented on source/_docs/installation.markdown:
```diff
@@ -131,3 +131,124 @@ For more details about `hass`, please refer to the [tools section](/docs/tools/h
+Add-Ons provide an easy way to add more functionality to your setup through the gui.
+Add-Ons in Hass.io are just docker containers containing some software.
+They are configured to for Home Assistant for easy setup.
+You always can add the functionaliy provided by an Add-On by yourself if you are **not** running Hass.io.
```
```suggestion
You always can add the functionality provided by an add-on by yourself if you are **not** running Hass.io.
```

frenck commented on source/_docs/installation.markdown:
```diff
@@ -131,3 +131,124 @@ For more details about `hass`, please refer to the [tools section](/docs/tools/h
+</div>
+
+This is a full appliance for running Home Assistant on your hardware using docker images running on HassOS.
+Home Assistant is running as a docker container as well as Add-Ons.
```
```suggestion
Home Assistant is running as a docker container as well as add-ons.
```

frenck commented on source/_docs/installation.markdown:
```diff
@@ -131,3 +131,124 @@ For more details about `hass`, please refer to the [tools section](/docs/tools/h
+
+This is a full appliance for running Home Assistant on your hardware using docker images running on HassOS.
+Home Assistant is running as a docker container as well as Add-Ons.
+You can add additional functionality through Add-Ons which are just preconfigured docker images to run with Home Assistant.
```
```suggestion
You can add additional functionality through add-ons which are just preconfigured docker images to run with Home Assistant.
```

frenck commented on source/_docs/installation.markdown:
```diff
@@ -131,3 +131,124 @@ For more details about `hass`, please refer to the [tools section](/docs/tools/h
+You can add additional functionality through Add-Ons which are just preconfigured docker images to run with Home Assistant.
+
+You will get:
+- You can use Add-Ons from the core team
```
```suggestion
- Access to the add-ons from the core team.
```

frenck commented on source/_docs/installation.markdown:
```diff
@@ -131,3 +131,124 @@ For more details about `hass`, please refer to the [tools section](/docs/tools/h
+
+You will get:
+- You can use Add-Ons from the core team
+- You can use Add-Ons from the community
```
```suggestion
- Access to the add-ons from the community.
```

frenck commented on source/_docs/installation.markdown:
```diff
@@ -131,3 +131,124 @@ For more details about `hass`, please refer to the [tools section](/docs/tools/h
+You will get:
+- You can use Add-Ons from the core team
+- You can use Add-Ons from the community
+- You can use any other Add-Ons
```
Considering the above 2...  what are the "other" add-ons?

frenck commented on source/_docs/installation.markdown:
```diff
@@ -131,3 +131,124 @@ For more details about `hass`, please refer to the [tools section](/docs/tools/h
+- You can use Add-Ons from the core team
+- You can use Add-Ons from the community
+- You can use any other Add-Ons
+- You can build your own Add-Ons (building and configuring docker images)
```
```suggestion
- To possibility to build your own add-ons.
```

frenck commented on source/_docs/installation.markdown:
```diff
@@ -131,3 +131,124 @@ For more details about `hass`, please refer to the [tools section](/docs/tools/h
+- You can use Add-Ons from the community
+- You can use any other Add-Ons
+- You can build your own Add-Ons (building and configuring docker images)
+- HassOS will run on the underling hardware
```
```suggestion
- HassOS will run on the underlying hardware.
```

frenck commented on source/_docs/installation.markdown:
```diff
@@ -131,3 +131,124 @@ For more details about `hass`, please refer to the [tools section](/docs/tools/h
+- You can use any other Add-Ons
+- You can build your own Add-Ons (building and configuring docker images)
+- HassOS will run on the underling hardware
+- You do/can not need to touch the operating system
```
```suggestion
- Automatic updates of the operating system (OTA).
```

frenck commented on source/_docs/installation.markdown:
```diff
@@ -131,3 +131,124 @@ For more details about `hass`, please refer to the [tools section](/docs/tools/h
+- You can build your own Add-Ons (building and configuring docker images)
+- HassOS will run on the underling hardware
+- You do/can not need to touch the operating system
+- You can update HassOS via Hass.io GUI
```
```suggestion
```

frenck reviewed (changes requested):
Left a partial review, which contains a trend you can apply to the rest of the PR.

Secondly, I agree with people above, this needs a separate page.

ajfriesen commented:
Will have a look later in the day.

frenck committed:
Apply suggestions from code review

Co-Authored-By: ajfriesen <ajfriesen@users.noreply.github.com>

ajfriesen committed:
Format markdown table

ajfriesen committed:
Remove statement

ajfriesen committed:
Add statement if docker knowledge is needed for the installations method

If you just started with home assistant and do not have any knowledge about linux and docker this make the decision easier for beginners.

ajfriesen committed:
Change the style as suggested by frenck

ajfriesen committed:
Move the installations methods to separate page

ajfriesen committed:
Remove withespace

ajfriesen committed:
Change title

ajfriesen committed:
Align heading because of the move to a dedicated page

ajfriesen committed:
Remove old block from installation.markdown

ajfriesen committed:
Add missing block to resolve conflict in installation.markdown

ajfriesen commented:
I totally agree to use a dedicated page and did just that.
Any ideas where it should be linked?

Sorry I got somehow confused with the branching and forking.
First time working on a community project on github.
Hope that I will learn more about the process :-)
